app/src/main/java/org/wikipedia/history/HistoryFragment.kt

```diff
@@ -86,6 +77,29 @@ class HistoryFragment : Fragment(), BackPressedHandler {
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.CREATED) {
+                launch {
+                    viewModel.uiState.collect {
+                        when (it) {
+                            is Resource.Success -> onLoadItemsFinished(it.data)
+                            is Resource.Error -> onError(it.throwable)
+                        }
+                    }
+                }
+                launch {
+                    viewModel.actionUiState.collect {
+                        when (it) {
+                            is Resource.Success -> onPagesDeleted()
```

Comment:
This will cause `onPagesDeleted()` to be called again on every screen rotation, even after an item was deleted. (it will keep showing a snackbar that says "0 items removed")

Reply:
(hint: `SingleLiveData`)

Reply:
Done!